### PR TITLE
feat: support show views statement

### PR DESCRIPTION
**Pull request by lyang24**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/4338
## What's changed and what's your intention?

__!!! DO NOT LEAVE THIS BLOCK EMPTY !!!__

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Added support for `SHOW VIEWS` SQL statements to display views in the database.
  - Users can now display views within a database using `SHOW VIEWS` and `SHOW VIEWS IN <database>` commands.

- **Tests**
  - Added test cases for `SHOW VIEWS` and `SHOW VIEWS IN <database>` statements to ensure correctness.

### CODE REVIEW

File: src/query/src/sql.rs [hunk 1]
```diff
@@ -725,6 +726,43 @@ pub fn show_create_view(
...
+        database
+    } else {
+        query_ctx.current_schema()
+    };
+
+    let projects = vec![(tables::TABLE_NAME, VIEWS_COLUMN)];
+    let filters = vec![
+        col(tables::TABLE_SCHEMA).eq(lit(schema_name.clone())),
+        col(tables::TABLE_CATALOG).eq(lit(query_ctx.current_catalog())),
+        col(tables::TABLE_TYPE).eq(lit("VIEW")),
```
Comment (lyang24) [inline src/query/src/sql.rs:746]: it would be great if we can replace the hard code here

File: src/query/src/sql.rs [hunk 2]
```diff
@@ -69,6 +70,7 @@ use crate::QueryEngineRef;
 const SCHEMAS_COLUMN: &str = "Database";
 const OPTIONS_COLUMN: &str = "Options";
 const TABLES_COLUMN: &str = "Tables";
+const VIEWS_COLUMN: &str = "VIEWS";
```
Comment (killme2008) [inline src/query/src/sql.rs:73]:
```suggestion
const VIEWS_COLUMN: &str = "Views";
```

File: src/query/src/sql.rs (hunk 1 above)
Comment (killme2008) [reply src/query/src/sql.rs:747]: 👍 
What about adding a function `name()` to `TableType` like:

https://github.com/GreptimeTeam/greptimedb/blob/main/src/catalog/src/information_schema/tables.rs#L270-L274

Then you can replace `VIEW` with `TableType::View.name()`.

Comment (killme2008) [reply src/query/src/sql.rs:747]: Another option is to query from `views`, the PR #4339 was merged.

File: src/sql/src/statements/show.rs [hunk 3]
```diff
@@ -200,6 +200,26 @@ impl Display for ShowCreateView {
...
+#[derive(Debug, Clone, PartialEq, Eq, Visit, VisitMut)]
+pub struct ShowViews {
+    pub kind: ShowKind,
+    pub database: Option<String>,
+}
+
+impl Display for ShowViews {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "SHOW ")?;
+        write!(f, "VIEWS")?;
```
Comment (killme2008) [inline src/sql/src/statements/show.rs:213]:
```suggestion
        write!(f, "SHOW VIEWS")?;
```

**Review by killme2008 — COMMENTED** [review]
Good job!

File: src/query/src/sql.rs (hunk 1 above)
Comment (lyang24) [reply src/query/src/sql.rs]: Thanks I used the views virtual table instead.

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**